src/rust/e2e-tests/src/test_utils/context.rs

```diff
@@ -65,36 +62,71 @@ impl AsyncTestContext for E2eTestContext {
     async fn setup() -> Self {
         let _guard = setup_tracing(SERVICE_NAME).expect("setup_tracing");
 
-        let event_source_client =
-            EventSourceServiceClient::connect_with_config(EventSourceClientConfig::parse())
-                .await
-                .expect("event_source_client");
-
-        let plugin_registry_client =
-            PluginRegistryServiceClient::connect_with_config(PluginRegistryClientConfig::parse())
-                .await
-                .expect("plugin_registry_client");
-
-        let graph_schema_manager_client =
-            GraphSchemaManagerClient::connect_with_config(GraphSchemaManagerClientConfig::parse())
-                .await
-                .expect("graph_schema_manager_client");
-
-        let pipeline_ingress_client =
-            PipelineIngressClient::connect_with_config(PipelineIngressClientConfig::parse())
-                .await
-                .expect("pipeline_ingress_client");
+        let event_source_client_config = Figment::new()
+            .merge(Env::prefixed("EVENT_SOURCE_CLIENT_"))
```

Comment:
I sort of don't love that every client of a given service has to re-state what its prefix is, but it's not a big deal. Nonetheless the Env::prefixed stuff is killer very cool

Reply:
This is just saying "hydrate the configuration from environment variables containing this prefix". So in this service (the e2e-tests service) the `event_source_client_config` will be hydrated by environment variables prefixed with `"EVENT_SOURCE_CLIENT_"`. This basically establishes a contract between the service's environment and the service. I think, therefore, it would be inappropriate to define this prefix somewhere else (e.g. on `ClientConfig` or on `EventSourceClient`. It's not up to those types what the prefix should be, it's up to e2e-tests.